### PR TITLE
Bug/form validation

### LINKED ISSUE #717: Wallet - Fix Bond Form validation issue

### PR DESCRIPTION
**Pull request by fmtabbara**

### CODE REVIEW

File: wallet-web/components/bond/BondNodeForm.tsx [hunk 1]
```diff
@@ -120,59 +134,49 @@ export default function BondNodeForm(props: TBondNodeFormProps) {
...
+      validVersion: validateVersion(version),
+      validMixPort: validateRawPort(mixPort),
+      validLocation:
+        props.type == NodeType.Gateway ? validateLocation(location) : true,
+      validVerlocPort:
+        props.type === NodeType.Mixnode ? validateRawPort(verlocPort) : true,
+      validHttpApiPort:
+        props.type === NodeType.Mixnode ? validateRawPort(httpApiPort) : true,
+      validClientsPort:
+        props.type === NodeType.Gateway ? validateRawPort(clientsPort) : true,
```
Comment (fmtabbara) [inline wallet-web/components/bond/BondNodeForm.tsx:169]: I'm just validating everything here including the values in the advanced area. If the advanced area is displayed then we validate the values that are supplied by the user. If the advanced area is not displayed then we are using the defaults from above which will be valid.

File: wallet-web/components/bond/BondNodeForm.tsx [hunk 2]
```diff
@@ -120,59 +134,49 @@ export default function BondNodeForm(props: TBondNodeFormProps) {
...
-    let validHost = validateHost(host)
-    let validVersion = validateVersion(version)
-
-    let validLocation =
-      props.type == NodeType.Gateway ? validateLocation(location) : true
-
+    verlocPort = DEFAULT_VERLOC_PORT.toString(),
+    mixPort = DEFAULT_MIX_PORT.toString(),
+    httpApiPort = DEFAULT_HTTP_API_PORT.toString(),
+    clientsPort = DEFAULT_CLIENTS_PORT.toString(),
```
Comment (fmtabbara) [inline wallet-web/components/bond/BondNodeForm.tsx:141]: If these values are undefined i.e. the advanced area is not displayed, then we fallback to our default values.

**Comment by fmtabbara** [comment]
I think the form could benefit from a little rewrite. What we have is perfectly fine and works well but we could reduce the overall code a little if we were to rewrite some of it. Maybe for discussion in sprint planning.

File: wallet-web/components/bond/BondNodeForm.tsx (hunk 2 above)
Comment (jstuczyn) [reply wallet-web/components/bond/BondNodeForm.tsx:141]: That sounds like a good approach

File: wallet-web/components/bond/BondNodeForm.tsx (hunk 1 above)
Comment (jstuczyn) [reply wallet-web/components/bond/BondNodeForm.tsx:169]: I guess this would simplify stuff

**Review by jstuczyn — APPROVED** [review]

**Comment by jstuczyn** [comment]
Feel free to rewrite whatever you think requires a rewrite. When I was writing this piece I knew it would be replaced relatively soon afterwards so I didn't put much thinking into it : )

**Comment by fmtabbara** [comment]
In general I think the form is fine. I'll just look at changing how the react state is being updated (in a separate PR). Everything else can pretty much stay the same.

**Review by neacsu — APPROVED** [review]
Bug is fixed after the patches

**Pull request merged**